src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/Maintainability/UseCrossPlatformIntrinsics.Fixer.cs

```diff
@@ -0,0 +1,262 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the MIT license.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using System.Threading;
+using System.Threading.Tasks;
+using Analyzer.Utilities;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Operations;
+
+namespace Microsoft.CodeQuality.Analyzers.Maintainability
+{
+    using static MicrosoftCodeQualityAnalyzersResources;
+    using RuleKind = UseCrossPlatformIntrinsicsAnalyzer.RuleKind;
+
+    public abstract class UseCrossPlatformIntrinsicsFixer : CodeFixProvider
+    {
+        public sealed override ImmutableArray<string> FixableDiagnosticIds { get; } = ImmutableArray.Create(
+            UseCrossPlatformIntrinsicsAnalyzer.RuleId
+        );
+
+        public sealed override FixAllProvider GetFixAllProvider()
+        {
+            // See https://github.com/dotnet/roslyn/blob/main/docs/analyzers/FixAllProvider.md for more information on Fix All Providers'
+            return WellKnownFixAllProviders.BatchFixer;
+        }
+
+        public sealed override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            SyntaxNode root = await context.Document.GetRequiredSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+            SyntaxNode node = root.FindNode(context.Span, getInnermostNodeForTie: true);
+
+            if (node is null)
+            {
+                return;
+            }
+
+            SemanticModel model = await context.Document.GetRequiredSemanticModelAsync(context.CancellationToken).ConfigureAwait(false);
+
+            if (model.GetOperation(node, context.CancellationToken) is not IInvocationOperation invocation)
```

Comment:
Is it necessary to get the IOperation here? Is syntax not enough?

Reply:
We use the invocation to get the arguments and avoid needing to deal with the syntax nuances between different kinds of invocations. It massively simplifies the rest of the logic

Reply:
Also note that you can add additional locations from the analyzer and simply retrieve them. That way we can just rely on syntax in the fixer.

Reply:
> seem to always get back to syntax again

Yes, because we need to do replacements and that generally needs syntax.

> Also note that you can add additional locations from the analyzer and simply retrieve them

This doesn't seem like it'd be simpler or cheaper, it'd just move unnecessary additional tracking into the diagnostic when that info is only needed if the fixer is applied. -- That is, the fixer needs to know things like which parameter is which, the diagnostic doesn't care.

Reply:
Note: it's fine to have the fixer to analysis like this.  though it's not ideal. Primarily because it means the user can see an analyzer error, butt hen no fix (because the fixer bailed out).

so the preference is *much more* that the analyzer do all work to ensure it absolutely knows it should be reported and can be fixed, and then the fixer  needs to do very litt.e.

Reply:
Notably, the diagnostic analyzer is already handling these cases and shouldn't be reporting diagnostics for cases that would bail.

However, it is still generally best practice to handle it defensively as it ensures that the JIT sees the expected invariants and can optimize the subsequent code accordingly. It also helps ensure that if any invariant changes, the analyzer isn't needlessly throwing (expensive) or crashing (something that many analyzers end up doing over time as possible syntax shapes change).